dev_docs/contributing/kibana_http_api_guidelines.mdx

```diff
@@ -1,820 +1,397 @@
+Internal HTTP APIs are only used by code Elastic owns, like the UI. By default, when you register a new HTTP API it will be classified as `internal`. For example, this API is internal:
 
-5. **Validation**: Validate input at creation/update time and return comprehensive errors.
+```typescript
+router.get({ path: '/api/foos' ... }, async () => {...})
+```
 
-#### Example: Complete API for a resource
+You can make an HTTP API public by changing the `access` setting to `public`:
 
 ```typescript
-// Retrieve 1 resource
-router.get(
-  {
-    path: '/api/resource/{id}',
-    validate: false, // `get` should not have bodies
-  },
-  async (context, request, response) => {
-    // Implementation...
-    return response.ok({ body: completeResourceState });
-  }
-);
-
-// Create, update
-router.(post|put)(
-  {
-    path: '/api/resource',
-    validate: {
-      body: schema.object({
-        name: schema.string(),
-        config: schema.object({...}),
-      }),
-    },
-  },
-  async (context, request, response) => {
-    // Implementation...
-    return response.ok({ body: completeResourceState });
-  }
-);
-
-// Delete
-router.delete(
-  {
-    path: '/api/resource/{id}',
-    validate: {
-      params: schema.object({
-        id: schema.string(),
-      }),
-    },
-  },
-  async (context, request, response) => {
-    // Implementation...
-    return response.ok();
-  }
-);
-
-// List many resources
-router.get(
-  {
-    path: '/api/resource',
-    validate: {
-      query: schema.object({
-        page: schema.maybe(schema.number()),
-        perPage: schema.maybe(schema.number()),
-      }),
-    },
-  },
-  async (context, request, response) => {
-    // Implementation...
-    return response.ok({
-      body: {
-        items: resources,
-        total: totalCount,
-        page: currentPage,
-        perPage: itemsPerPage
-      }
-    });
-  }
-);
+router.get({ path: '/api/foos', options: { access: 'public' } }, async () => {...})
 ```
 
-#### API imposed challenges
+See the principle regarding [commitment](#commitment) before making APIs public!
 
-API design decisions can create real challenges for Terraform resource implementation. The Elasticsearch [Index Settings API](https://www.elastic.co/docs/api/doc/elasticsearch/operation/operation-indices-put-settings) (\`PUT /\{index\}/\_settings\`) treats static settings (which require index recreation) and dynamic settings (which can be updated in place) identically. This creates a complex situation forcing the implementation to handle all settings together because the API doesn't offer a way to distinguish them upfront.
+## Structure and conventions
 
-```bash
-{
-  "error": {
-    "reason": "Can't update non dynamic settings [[index.codec, index.number_of_shards]] for open indices"
-}
-```
+The following are a quick reference for how to structure your HTTP APIs.
 
-[Index settings update bug](https://github.com/elastic/terraform-provider-elasticstack/issues/52)
+### RESTful-like APIs
```

Comment:
Please weigh in here, is this point helpful? Shall I remove it?

Reply:
It's useful in the sense that it gives contributors a base spec to fall back to if they're completely lost about how to start.